### DISCUSSION TITLE
Linearization method in Pybamm

### DISCUSSION THREAD

**Discussion by xyyrly** [pybamm-team/PyBaMM#4014] [Q&A] [answered]
Hi team,

For DFN model, I am wondering how Pybamm linearizes the strongly non-linear system. I try to develop a microscale lithium-ion battery model and use Newton method to do linearization, but the solution for potential is weird. Can I ask what root-finding algorithm is used in Pybamm? 

Thank u very much!

**Comment by valentinsulzer** [comment, marked as answer]
We use Sundials (either via CasADi or the IDA KLU solver directly) to solve the non-linear DAE system

**Reply by xyyrly** [reply]
Thank u for your reply! That will help a lot!
